packages/ra-core/src/util/useFieldValue.spec.tsx

```diff
@@ -0,0 +1,64 @@
+import * as React from 'react';
+import { render, screen } from '@testing-library/react';
+import { useFieldValue, UseFieldValueOptions } from './useFieldValue';
+import { RecordContextProvider } from '../controller';
+import { SourceContextProvider } from '..';
+
+describe('useFieldValue', () => {
+    const Component = (props: UseFieldValueOptions) => {
+        return <div>{useFieldValue(props) ?? 'None'}</div>;
+    };
+
+    it('should return undefined if no record is available', async () => {
```

Comment:
you're missing a test about deep sources

Reply:
Added